lib/ansible/executor/task_executor.py

```diff
@@ -770,6 +763,31 @@ def _evaluate_failed_when_result(result):
         display.debug("attempt loop complete, returning result")
         return result
 
+    @staticmethod
+    def _retries_requested(_task: Task) -> boolean:
+        return bool((_task.retries and _task.retries > 1) or _task.until)
+
+    @staticmethod
+    def process_retry_parameters(retries_requested: bool, _task: Task) -> Tuple[int, int, Union[str, List]]:
+        '''
+        Extract the parameters used in retrying the task
+        '''
+        attempts = 1
+        if retries_requested:
+            if _task.retries:
+                attempts += _task.retries
+            else:
+                attempts += 3
+
+        if attempts <= 0:  # guard against negative retries
+            attempts = 1
+
+        delay = _task.delay
+        if delay < 0:
```

Comment:
The reason I asked for integration tests is because it's easy to check backwards compatibility. I can't copy your unit tests onto devel without also editing code since they use the new method.